### PR TITLE
Ens subdomains registration

### PR DESCRIPTION
**Pull request by namesty**
This PR includes necessary code for:

- Swap web3 for ethers and its wrapping class, taken from Web3API's monorepo

- ENS subdomain registration under "web3api.eth" domain
- ContentHash saving

- Domain owner resolution
- Domain contentHash resolution

Steps for testing:

- Connect wallet to ropsten
- Fill IPFS input
- Fill subdomain input
- Sign the 3 transaction prompts (register subdomain, set resolver, set content hash)

After a successful registration:

- Notice that when writing a new subdomain, the input field will show in red if the subdomain is already taken
- I didn't see where to output the content hash so I left that service unused for now but it's working, you can verify it with a simple invocation and console.log

### CODE REVIEW

File: packages/ui/src/components/SignInArea.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ const SignInArea = ({ onDark }: SignInAreaProps) => {
     setShowDisconnectModal(true)
   }
   const handleSignIn = () => {
-    setShowSignInModal(true)
+    setShowConnectModal(true)
```
Comment (jeffscottward) [inline packages/ui/src/components/SignInArea.tsx:32]: this is wrong - "Signin" is for Web2 stuff
we use the term "Connect" for web3 stuff.
There should be no connecting to web3 until you are on the publish page - hence the messaging on the modal that corresponds.
Please ask andrew if you have UX questions

File: packages/ui/src/utils/ethereum.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
+import { ethers } from "ethers"
+
```
Comment (jeffscottward) [inline packages/ui/src/utils/ethereum.ts:2]: We should not use ES6 classes anywhere. 
https://reactjs.org/docs/hooks-intro.html#classes-confuse-both-people-and-machines

Please convert.

File: packages/ui/src/utils/ethereum.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    address: Address,
+    method: string,
+    args: (string | Record<string, string | number> )[]
+  ): Promise<string> {
+    const contract = this.getContract(address, [method]);
+    const funcs = Object.keys(contract.interface.functions);
+    try {
+      const tx = await contract[funcs[0]](...args);
+      const res = await tx.wait();
+      // TODO: improve this
```
Comment (jeffscottward) [inline packages/ui/src/utils/ethereum.ts:119]: whats needed here?

File: packages/ui/src/components/SignInArea.tsx (hunk 1 above)
Comment (jeffscottward) [reply packages/ui/src/components/SignInArea.tsx:32]: please return back to original setShowSignInModal and utilize signing in at the point of need on publish page when clicking Register ENS

File: packages/ui/src/utils/ethereum.ts (hunk 2 above)
Comment (namesty) [reply packages/ui/src/utils/ethereum.ts:2]: Ahh, but these are typescript classes, which do not contain the headaches of ES6 classes (well described in the article you linked).

The purpose of this class is to wrap the Ethers/Web3 class, so usage would be similar to how you would use the Web3 class or ethers class.

We could definitely refactor this into something else if you prefer though!

File: packages/ui/src/utils/ethereum.ts (hunk 3 above)
Comment (namesty) [reply packages/ui/src/utils/ethereum.ts]: Was wondering what to return. If only the TX Hash or the whole TX object. I opted for the latter

File: packages/ui/src/components/SignInArea.tsx [hunk 4]
```diff
@@ -35,7 +36,7 @@ const SignInArea = ({ onDark }: SignInAreaProps) => {
     setShowSignOutModal(true)
   }
 
-  console.log({dapp})
+  console.log({ dapp })
```
Comment (jeffscottward) [inline packages/ui/src/components/SignInArea.tsx:39]: Dump this

File: packages/ui/src/utils/ethereum.ts (hunk 2 above)
Comment (jeffscottward) [reply packages/ui/src/utils/ethereum.ts:2]: Oh I get the intent, that's fine, but it sticks out from the rest of the code base. 

There's overhead for having to remember the properties of this approach and how it maps back to actually what it represents in JS.

Function composition > syntax sugar.

Let's change it please.

**Review by jeffscottward — CHANGES_REQUESTED** [review]
Refactor eth wrap and drop a console log. Assuming it works, I'm just reviewing from my phone right now.

**Review by jeffscottward — APPROVED** [review]

**Pull request merged**